### DISCUSSION TITLE
Limiting direct access to files in a public bucket behind a CDN

### DISCUSSION THREAD

**Discussion by binaryfire** [minio/minio#20552] [Q&A] [answered]
Hi all

I'm using Bunny CDN to cache files I'm serving from a public bucket. Is there a policy I can use to restrict access to just the CDN's servers? I realise it won't be secure and that's ok - the files are public. I just need to discourage hotlinking directly to files to minimise egress bandwidth as much as possible.

**Comment by klauspost** [comment, marked as answer]
https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucket-policies.html

https://min.io/docs/minio/linux/administration/identity-access-management/policy-based-access-control.html#supported-s3-policy-condition-keys

You can use `aws:SourceIp` (if known), or `aws.UserAgent` (if you can modify). Both can do a white-list. User-Agent would probably be the easiest, if you can modify it on the CDN.

**Reply by binaryfire** [reply]
@klauspost Perfect, thank you!
